test/sym/global.wat

```diff
@@ -4,28 +4,46 @@
   (import "symbolic" "f32_symbol" (func $f32_symbol (result f32)))
   (import "symbolic" "f64_symbol" (func $f64_symbol (result f64)))
 
-  (global $x (mut i32) (i32.const 0))
-  (global $y (mut i64) (i64.const 0))
-  (global $i (mut f32) (f32.const 0))
-  (global $j (mut f64) (f64.const 0))
+  (global $w (mut i32) (i32.const 0))
+  (global $x (mut i64) (i64.const 0))
+  (global $y (mut f32) (f32.const 0))
+  (global $z (mut f64) (f64.const 0))
 
-  (func $start
-    (global.set $x (call $i32_symbol))
-    (global.set $y (call $i64_symbol))
-    (global.set $i (call $f32_symbol))
-    (global.set $j (call $f64_symbol))
+  (func $i32 (param $a i32)
+    (if (i32.eqz (local.get $a))
+      (then unreachable))
+  )
 
-    (if (i32.lt_s (i32.const 5) (global.get $x))
+  (func $i64 (param $a i64)
+    (if (i64.eqz (local.get $a))
       (then unreachable))
+  )
 
-    (if (i64.gt_s (i64.const 5) (global.get $y))
+  (func $f32 (param $a f32)
+    (if (f32.eq (local.get $a) (f32.const 0))
       (then unreachable))
+  )
 
-    (if (f32.lt (f32.const 5) (global.get $i))
+  (func $f64 (param $a f64)
+    (if (f64.eq (local.get $a) (f64.const 0))
       (then unreachable))
+  )
+
+  (func $start
+    ;; objective: check that interactions with globals are ok
+
+    (global.set $w (call $i32_symbol))
+    (global.set $x (call $i64_symbol))
+    (global.set $y (call $f32_symbol))
+    (global.set $z (call $f64_symbol))
 
-    (if (f64.gt (f64.const 5) (global.get $j))
-      (then unreachable)))
+    ;; 'unreachable' when argument is equal to 0
+    ;; in global.t, only one model symbol is equal to 0
+    (call $i32 (global.get $w))
+    (call $i64 (global.get $x)) ;; (symbol_0) w != 0
+    (call $f32 (global.get $y)) ;; (symbol_0) w != 0 AND (symbol_1) x != 0
+    (call $f64 (global.get $z)) ;; (symbol_0) w != 0 AND (symbol_1) x != 0 AND (symbol_2) y != 0
+  )
```

Comment:
I believe a better thing to test is the following:

```ocaml
local.set $x (call sym_i32)
global_set $g (local.get $x)
assert (i32.eq (local.get $x) (global.get $x)
```

i.e. storing something in a global and then reading it should give the same value